Sources/MapboxCoreNavigation/CoreNavigationNavigator.swift

```diff
@@ -44,6 +44,14 @@ class Navigator {
     
     static func _recreateNavigator() { _navigator = .init() }
     
+    /**
+     Profile setting, used for selecting tiles type for navigation.
+     
+     This property can only be modified before creating `Navigator` shared instance, all
+     further changes to this property will have no effect. Defaults to `automobile`.
+     */
+    static var datasetProfile: ProfileIdentifier = .automobile
```

Comment:
Could it be placed into `NavigationSettings`? So that we don't have to provide it in constructors of `PassiveLocationManager` and `RouteController`.

Reply:
This setting should mirror `NavigationOptions.datasetProfile`, so having a parallel value in `NavigationSettings` will be confusing. And we cannot remove it from "options" too. See [this thread](https://github.com/mapbox/mapbox-navigation-ios/pull/3717#discussion_r796439137) for more details about exposing it through constructors.